File: pa/pa3/DictionaryTest.java
/* DictionaryTest.java
 * Juan Andreas
 * jandreas
 * 12b
 * Test for Dictionary
 */

public class DictionaryTest{

   public static void main(String[] args){
      Dictionary D = new Dictionary();

      System.out.println("This is the original list: ");
      D.insert("1","z");
      D.insert("2","y");
      System.out.println(D);
      //empties linked list
      D.makeEmpty();
      //refills list
      D.insert("1","a");
      D.insert("2","b");
      D.insert("3","c");

      System.out.println("This is the new list: ");
      System.out.println(D);
      //finds matching values to keys
      System.out.println("These are the values of the new list keys: ");
      System.out.println(D.lookup("1"));
      System.out.println(D.lookup("2"));
      System.out.println(D.lookup("3"));
      System.out.println();

      D.delete("3");

      System.out.println("This is the final list: ");
      System.out.print(D);
      System.out.println();
      System.out.println("List empty?");
      System.out.println(D.isEmpty());
   }
}

File: pa/pa5/DictionaryTest.c
/* Juan Andreas
 * jandreas
 * 12B
 * 5/28/2016
 * Dictionary Test file
 */
#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h"
 
#define MAX_LEN 180

int main(int argc, char* argv[]){
   Dictionary A = newDictionary();
   char* k;
   char* v;
   char* word1[] = {"one","two","three","four"};
   char* word2[] = {"dog","cat","chicken","cow"};
   int i;

   for(i=0; i<4; i++){
      insert(A, word1[i], word2[i]);
   }
   printf("This is the original list: \n");
   printDictionary(stdout,A);
   
   for(i=0; i<4; i++){
      k = word1[i];
      v = lookup(A, k);
      printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
   }
   
   delete(A, "one");
   delete(A, "two");
   
   printf("\nThis is the edited list: \n");
   printDictionary(stdout, A);

   for(i=0; i<4; i++){
      k = word1[i];
      v = lookup(A, k);
      printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
   }
   
   printf("\n");
   printf("%s\n", (isEmpty(A)?"true":"false"));
   printf("%d\n", size(A));
   makeEmpty(A);
   printf("%s\n", (isEmpty(A)?"true":"false"));
   
   freeDictionary(&A);
   
   return(EXIT_SUCCESS);
}
